parquet-hadoop/src/main/java/org/apache/parquet/hadoop/metadata/ColumnChunkMetaData.java

```diff
@@ -165,10 +200,10 @@ protected static boolean positiveLongFitsInAnInt(long value) {
     return (value >= 0) && (value + Integer.MIN_VALUE <= Integer.MAX_VALUE);
   }
 
-  private final EncodingStats encodingStats;
+  protected EncodingStats encodingStats;
```

Comment:
These are accessed only inside the same class so you may keep them `private`.

Reply:
these fields are accessed by the `EncryptedColumnChunkMetaData` class, that sets their values after decrypting the column metadata.

Reply:
Yep, but `EncryptedColumnChunkMetaData` is defined inside the class `ColumnChunkMetaData`. It makes them access each other's `private` members. Even the class`EncryptedColumnChunkMetaData` can be declared `private` if you don't want to use the type outside of `EncryptedColumnChunkMetaData`.
See [Nested Classes](https://docs.oracle.com/javase/tutorial/java/javaOO/nested.html) for details.

Reply:
hmm, when I change `encodingStats` to `private`, I get a compilation error  
_The field ColumnChunkMetaData.encodingStats is not visible_
at line 619,
`this.encodingStats = shadowColumnChunkMetaData.encodingStats;`

Reply:
OK, I got it now. So, these are not nested classes. There are 4 different classes next to each other in the same java file. This is valid since only one class is public but I've never seen such design in production. I've never realized this design in `ColumnChunkMetaData`. There is no reason why one would put multiple classes in the same file but not nesting them.

So, without refactoring this structure I'll accept `encodingStats` to not being `private`. But, we can narrow the visibility by using _package private_ (no modifier).

Reply:
Sure, will change this.